toolchain/check/handle_function.cpp

```diff
@@ -342,26 +340,28 @@ static auto HandleFunctionDefinitionAfterSignature(
   CheckFunctionReturnType(context, function.return_slot_id, function,
                           SemIR::SpecificId::Invalid);
 
+  auto params_to_complete =
+      context.inst_blocks().GetOrEmpty(function.calling_convention_params_id);
+  if (function.return_slot_id.is_valid()) {
+    // Exclude the return slot because it's diagnosed above.
+    params_to_complete = params_to_complete.drop_back();
+  }
   // Check the parameter types are complete.
-  for (auto param_ref_id : llvm::concat<const SemIR::InstId>(
-           context.inst_blocks().GetOrEmpty(function.implicit_param_refs_id),
-           context.inst_blocks().GetOrEmpty(function.param_refs_id))) {
+  for (auto param_ref_id : params_to_complete) {
     if (param_ref_id == SemIR::InstId::BuiltinError) {
       continue;
     }
-    auto param_info =
-        SemIR::Function::GetParamFromParamRefId(context.sem_ir(), param_ref_id);
```

Comment:
I see you removing several references to `GetParamFromParamRefId`, is it still in use?